certora/specs/SocialRecoveryModule.spec

```
@@ -1,3 +1,27 @@
-rule demo() {
-    assert true;
+using SafeHarness as safeContract;
+
+methods {
+    // Safe Functions
+    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+    function safeContract.disableModule(address, address) external;
+}
+
+// A setup function that requires Safe contract to enabled the Social Recovery
+// Module.
+function setupRequireRecoveryModule() {
+    require(safeContract.isModuleEnabled(currentContract));
+}
+
+// This is a dummy rule to verify the Safe Contract Setup with the Social Recovery
+// Module is working as intended.
+rule recoveryModuleCanBeDisabled {
+    env e;
+    address prevModule;
+
+    setupRequireRecoveryModule();
+
+    safeContract.disableModule@withrevert(e, prevModule, currentContract);
+    bool isReverted = lastReverted;
+
+    assert isReverted || (!isReverted && !safeContract.isModuleEnabled(currentContract));
```

Personally, I am a fan of the "implies" construction:

```suggestion
    assert !isReverted => !safeContract.isModuleEnabled(currentContract);
```

This is, in English, "disable module did not revert implies that the module is no longer enabled".

---

Done 👍🏾